### PR TITLE
Wait until popup message disappears or die if there is no timeout

### PR DESCRIPTION
**Pull request by kravciak**
Fix for:
https://openqa.suse.de/tests/993577
https://openqa.suse.de/tests/1012726#step/installation/6

Local run: http://dhcp91.suse.cz/tests/6046

### CODE REVIEW

**Comment by sysrich** [comment]
-1 : If we're avoiding a 'known issue', it should be a softfail

If we accepted this PR as is we could potentially skip past thousands of REALLY NASTY autoyast bugs

**Comment by kravciak** [comment]
My idea was:
 - for bugs create workaround needles
 - for features create normal needles matching the message

OK is pressed only if warning matches known needle - please explain more, I don't understand how we can miss bugs here
Current run: https://openqa.suse.de/tests/994647#step/installation/2

**Comment by sysrich** [comment]
https://openqa.suse.de/tests/994647#step/installation/2

There is a 9 second timeout on that message (see just below mouse cursor)

And yet this PR and https://openqa.suse.de/tests/993577 suggests that the current behaviour is that the timeout is not functioning and and message will remain even after the timeout has completed

The behaviour I describe above is a _bug_ "Autoyast Timeout/Automatic skip is not functioning"

And the very sort of bug which this PR would ignore, as shown by https://openqa.suse.de/tests/994647#step/installation/2

**Comment by kravciak** [comment]
I asked YaST developers about this and response was: it depends on the skelcd-control-SLES, in the profile you can specify how to handle warnings and errors. so if could a block in the installation or not  depending on what the user define.

I will report a bug for that to double-check. Is this PR acceptable if they close it as feature?

**Comment by sysrich** [comment]
If they close it as a feature, I'd rather see caasp have a bug to change
it's skelcd to act in the way you want to here, rather than changing the
tests to ignore the behaviour defined in the product at the moment

It seems very wrong to me to have the timeout enabled, as caasp does right
now, but then not skip at the end of the timeout, which is the usual
behaviour for timeouts in autoyast

**Comment by dzedro** [comment]
* If in previous test we use HDDSIZEGB=30 and in following autoyast_reinstall we use HDDSIZEGB=22 then we have to expect this message and handle it so.
* No idea if the pop-up should have timeout or not, don't know if autoyast messages have some criticality and based on that they timeout after while or not.
* I like Martins idea that known messages that have needle will pass and unknown will fail, existing [fail](https://openqa.suse.de/tests/1002616) is just broken. Maybe handle/record warning differently for messages with or without timeout ? Btw until 'warning-pop-up' was added messages with timeout were invisible in test.

**Comment by kravciak** [comment]
Fixed, please check

**Comment by kravciak** [comment]
https://bugzilla.suse.com/show_bug.cgi?id=1045470

File: tests/autoyast/installation.pm [hunk 1]
```diff
@@ -118,9 +118,11 @@ sub run {
...
         }
         elsif (match_has_tag('warning-pop-up')) {
-            unless (check_screen('autoyast-known-warning', 0)) {
-                die 'Check the pop-up message, it is not critical so installation would continue';
-            }
+            die "Unknown popup message" unless check_screen('autoyast-known-warning', 0);
+            # Die if there is no timeout
+            die 'Popup message without timeout' unless wait_screen_change(undef, 5);
+            # Wait until popup disappears
+            sleep 3 while check_screen('autoyast-known-warning', 0);
```
Comment (dzedro) [inline tests/autoyast/installation.pm]: Why while and not if ? Can't this get infinite sleep loop ?

Comment (kravciak) [inline tests/autoyast/installation.pm]: I can put bigger sleep (10 to be safe) there without "while" check

**Pull request merged**